### PR TITLE
Support Payment sandbox mode

### PR DESCRIPTION
**Pull request by skyred**
Following discussion from https://github.com/overtrue/wechat/issues/507

This PR adds the feature. So, app user can set sandbox mode like this:
```
$app = new Application($options);
$wechat_pay = $app->payment;
$wechat_pay->enableSandbox(TRUE);
```

Another option to enable to sandbox can be in `$option`:

```
    $options = [
      // 前面的appid什么的也得保留哦
      'app_id' => $appid,
      // ...
      // payment
      'payment' => [
        'merchant_id'        => $mch_id,
        'key'                => $key,
        'cert_path'          => 'path/to/your/cert.pem', // XXX: 绝对路径！！！！
        'key_path'           => 'path/to/your/key',      // XXX: 绝对路径！！！！
        'notify_url'         => $payment_gateway_plugin->getNotifyUrl()->toString(),
         'device_info'     => '013467007045764',
         'sub_app_id'      => '',
         'sub_merchant_id' => '',
         'sandbox_mode' => TRUE,
        // ...
      ],
    ];
```
But, this PR does't include the second way. Further contribution is welcomed.

### CODE REVIEW

File: src/Payment/API.php [hunk 1]
```diff
@@ -68,9 +78,12 @@ class API extends AbstractAPI
      *
      * @param \EasyWeChat\Payment\Merchant $merchant
      */
-    public function __construct(Merchant $merchant)
+    public function __construct(Merchant $merchant, $isSandboxMode = FALSE)
     {
         $this->merchant = $merchant;
+        if ($isSandboxMode == TRUE) {
```
Comment (overtrue) [inline src/Payment/API.php]: `TRUE` -> `true`

Comment (overtrue) [inline src/Payment/API.php]:
```diff
- if ($isSandboxMode == TRUE) {
+ if ($isSandboxMode) {
```

File: src/Payment/API.php [hunk 2]
```diff
@@ -38,18 +38,28 @@ class API extends AbstractAPI
      */
     protected $merchant;
 
+    /**
+     * Sandbox box mode
+     *
+     * @var string
+     */
+     protected $sandboxMode = '';
```
Comment (overtrue) [inline src/Payment/API.php]:
```diff
- protected $sandboxMode = '';
+ protected $prefix = '';
```

**Comment by skyred** [comment]
It looks like PaymentAPITest.php has failed. https://travis-ci.org/overtrue/wechat/jobs/224199863#L342

**Comment by overtrue** [comment]
image

File: tests/Payment/PaymentAPITest.php [hunk 3]
```diff
@@ -240,7 +240,7 @@ public function testAuthCodeToOpenId()
 
         $response = $api->authCodeToOpenId('authcode');
 
-        $this->assertEquals(API::API_AUTH_CODE_TO_OPENID, $response['api']);
+        $this->assertEqualsself::API_ENDPOINT . $this->prefix . API::API_AUTH_CODE_TO_OPENID, $response['api']);
```
Comment (overtrue) [inline tests/Payment/PaymentAPITest.php]: 请在本地测试通过再提交，谢谢。

**Comment by skyred** [comment]
Let me close this PR, and do a clean one.

**Comment by mingyoung** [comment]
@skyred  沙箱模式的签名是不是要从微信服务器获取？

**Pull request closed without merging**